File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        dic = {}
        dic[numbers[0]] = 0
        for i in range(1,len(numbers)):
            comp = target - numbers[i]
            if comp in dic:
                return [dic[comp]+1,i+1]
            else:
                dic[numbers[i]] = i
        return [-1,-1]

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer,Integer> hm = new HashMap<>();
        hm.put(numbers[0],0);
        for(int i=1;i<numbers.length;i++){
            int compliment = target - numbers[i];
            if (hm.containsKey(compliment)){
                return new int[]{hm.get(compliment)+1,i+1};
            }
            else{
                hm.put(numbers[i],i);
            }
        }
        return new int[]{-1,-1};
    }
}